src/main/webapp/js/index.js

```diff
@@ -13,36 +13,29 @@
 // limitations under the License.
 
 
-/** Fetch receipts from the server and add them to the DOM. */
+/** Fetches receipts from the server and adds them to the DOM. */
 async function searchReceipts() {
   const label = document.getElementById('search-input').value;
   const response = await fetch(`/search-receipts?label=${label}`);
   const receipts = await response.json();
 
-  clearExistingReceipts();
+  clearExistingDisplay();
   displayReceipts(label, receipts);
 }
 
 /**
- * Clear out existing receipts display.
- *
- * Note: 3 childNodes make up a receipt card.
- * The first 3 nodes are the template receipt card and each subsequent
- * 3 are clones of the template. The template must never be deleted or
- * it'll be unusable in the future.
+ * Clears out receipts display including old receipts and error messages.
  */
-function clearExistingReceipts() {
-  const receiptsDisplay = document.getElementById('receipts-display');
-  const numChildNodes = receiptsDisplay.childNodes.length;
-
-  // Make sure not to delete first 3 childNodes
-  for (let i = numChildNodes; i > 3; i--) {
-    receiptsDisplay.removeChild(receiptsDisplay.lastChild);
-  }
+function clearExistingDisplay() {
+  const elements = document.querySelectorAll(
+      '.col-md-6, .col-md-12.text-center.error-message');
```

Comment:
I would add a comment or have this string be in a descriptive variable. Otherwise it's not clear why you are querying for this.